src/Npgsql.EntityFrameworkCore.PostgreSQL/Migrations/Internal/NpgsqlHistoryRepository.cs

```diff
@@ -90,17 +90,27 @@ public override string GetCreateIfNotExistsScript()
 
         public override string GetBeginIfNotExistsScript(string migrationId)
         {
-            throw new NotSupportedException("Generating idempotent scripts for migration is not currently supported by Npgsql");
+            return @"
+DO $$
+BEGIN
+    IF NOT EXISTS(SELECT 1 FROM public.""__EFMigrationsHistory"" where ""MigrationId"" = '{migrationId}') THEN"
+                    .Replace("{migrationId}", migrationId);
```

Comment:
Ahh, welcome to a wonderful new world which will vastly improve your code quality :)

It's a C# 6.0 feature (VS2015), so it's been around for a while and is perfectly safe to use.